token/program-2022/src/extension/confidential_mint_burn/processor.rs

```diff
@@ -95,8 +96,8 @@ fn process_rotate_supply_elgamal_pubkey(
     let authority_info = next_account_info(account_info_iter)?;
     let authority_info_data_len = authority_info.data_len();
 
-    let authority = Option::<Pubkey>::from(mint_burn_extension.authority)
-        .ok_or(TokenError::NoAuthorityExists)?;
+    let authority = OptionalNonZeroPubkey::try_from(mint_authority)?;
+    let authority = Option::<Pubkey>::from(authority).ok_or(TokenError::NoAuthorityExists)?;
```

Comment:
Rather than do two conversions, we can use `ok_or` on the mint_authority directly: https://github.com/solana-labs/solana-program-library/blob/4ff64ce961f8c4f0156a7a8408e9c10a48385b0d/token/program-2022/src/pod.rs#L219
```suggestion
    let authority = mint.base.mint_authority.ok_or(TokenError::NoAuthorityExists)?;
```